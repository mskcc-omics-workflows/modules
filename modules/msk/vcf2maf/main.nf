
process VCF2MAF {
    tag "$meta.id"
    label 'process_low' 

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'ghcr.io/msk-access/vcf2maf:1.6.21-vep105':
        'ghcr.io/msk-access/vcf2maf:1.6.21-vep105' }"

    input:

    tuple val(meta), path(vcf), path(ref_fasta)
    
    output:
    
    tuple val(meta), path("*.maf"), emit: maf


    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def args2 = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
   
    """
    vcf2maf.pl $args \\
    --normal-id ${meta.control_id} \\
    $args2 --tumor-id \\
    ${meta.case_id} \\
    --vcf-normal-id ${meta.control_id} \\
    --vcf-tumor-id ${meta.case_id} \\
    --input-vcf ${vcf} \\
    --output-maf ${prefix}.maf \\
    --ref-fasta ${ref_fasta} \\
    --vep-path /usr/local/bin


    cat <<-END_VERSIONS > versions.yml
    "${task.process}": 
        vcf2maf: \$(echo \$(vcf2maf.pl --help) | echo 'vcf2maf version 1.6.21 VEP v105')
    END_VERSIONS
    """
    
    stub:
    def args = task.ext.args ?: ''
    def args2 = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    echo "stub test" >> ${prefix}.maf

    cat <<-END_VERSIONS > versions.yml
    "${task.process}": 
        vcf2maf: \$(echo \$(vcf2maf.pl --help) | echo 'vcf2maf version 1.6.21 VEP v105')
    END_VERSIONS
    """
}


//standard VCF2MAF args arguments: --buffer-size 5000 --maf-center mskcc.org --min-hom-vaf 0.7 --ncbi-build GRCh37 
//standard VCF2MAF args2 arguments: --retain-info set,TYPE,FAILURE_REASON,MUTECT 