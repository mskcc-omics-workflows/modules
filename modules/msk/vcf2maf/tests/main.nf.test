nextflow_process {

    name "Test Process VCF2MAF"
    script "../main.nf"
    process "VCF2MAF"

    tag "modules"
    tag "modules_nfcore"
    tag "vcf2maf"


    test("normal_tumor_sample1 - stub") {

        options "-stub"

        when {
            process {
                """
                
                input[0] = [
                    [ id:'tumor.normal', case_id:'tumor', control_id:'normal' ], // meta map
                    file("stub_vcf_file"),
                    file("stub_fasta_file")
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
