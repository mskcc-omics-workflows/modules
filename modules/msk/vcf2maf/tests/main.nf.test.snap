{
    "normal_tumor_sample1 - stub": {
        "content": [
            {
                "0": [
                    
                ],
                "maf": [
                    
                ]
            }
        ],
        "timestamp": "2024-02-29T16:55:00.115751"
    },
    "normal_tumor_sample1": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "tumor.normal",
                            "case_id": "tumor",
                            "control_id": "normal"
                        },
                        "tumor.normal.maf:md5,43f84d3fc1e864e9a0a192e2e2765a23"
                    ]
                ],
                "maf": [
                    [
                        {
                            "id": "tumor.normal",
                            "case_id": "tumor",
                            "control_id": "normal"
                        },
                        "tumor.normal.maf:md5,43f84d3fc1e864e9a0a192e2e2765a23"
                    ]
                ]
            }
        ],
        "timestamp": "2024-02-28T13:05:35.448181"
    }
}